### PR TITLE
v2: refactor code structure

### PR DESCRIPTION
**Pull request by quietbits**
Frontend and backend are built separately now. Most changes are for moving files around + configs.

I couldn't get `lint-staged` working, so leaving that for later.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,16 +14,19 @@
...
+    "start:backend": "cd backend && yarn start && cd ..",
+    "clean:frontend": "cd frontend && yarn clean && cd ..",
+    "clean:backend": "rm -rf node_modules && rm -rf build && cd backend && yarn clean && cd ..",
+    "clean": "run-p clean:frontend clean:backend",
+    "prebuild:frontend": "cd frontend && yarn install && cd ..",
+    "prebuild:backend": "cd backend && yarn install && cd ..",
+    "prebuild": "run-s prebuild:backend prebuild:frontend",
+    "build:frontend": "cd frontend && yarn build && cd ..",
+    "build:backend": "cd backend && yarn build && cd ..",
+    "build": "run-s build:frontend build:backend"
```
Comment (piyalbasu) [inline package.json:29]: is there a reason we're not using `workspaces`? It would simplify a lot of these build steps

**Review by piyalbasu — APPROVED** [review]

**Review by acharb — APPROVED** [review]
lgtm!

only thing is the backend tests aren't part of the workflow anymore, and the yarn test command was removed. Not critical, but we'd probably want to include that (maybe after all your v2 frontend changes so doesn't slow down your workflow) to be safe with next backend changes. I added a ticket to track https://github.com/stellar/dashboard/issues/245

File: package.json (hunk 1 above)
Comment (quietbits) [reply package.json:29]: It took us a while to figure out how to build frontend and backend properly. Adding an extra variable in config seemed a bit too much. :sweat_smile: We should definitely try workspaces down the line. :100:

**Comment by quietbits** [comment]
@acharb, thanks for opening an issue for tests. We'll definitely fix them later. I had weird timing issues running tests with GitHub Actions, but new build process might have fixed it. :crossed_fingers:

**Pull request merged**